odoo/addons/base/models/res_currency.py

```diff
@@ -122,6 +123,16 @@ def _num2words(number, lang):
                         )
         return amount_words
 
+    def amount_to_string(self, amount, human_readable=False):
```

Comment:
Why not using amount_to_text ?

Reply:
@tde-banana-odoo Because amount_to_text convert text in "letter"
```
>>> env["res.currency"].search([])[0].amount_to_text(555)
'Five Hundred And Fifty-Five Euros'
```

Reply:
Then maybe use format_amount
```
def format_amount(env, amount, currency, lang_code=False)
```

You could add a parameter for the "humanize" parameter (we should find a better name like "decimalize" ... mmh ...).

Reply:
Finally move `format_human_readable_amount` of digest in `tools`

Reply:
(and improve it to show the currency symbol)